### PR TITLE
Add dockerization of DKG app

### PR DESCRIPTION
**Pull request by cthoyt**
This PR makes updates to the domain knowledge graph (DKG) construction and web interface. It also contains docker settings for running it (though they're not _quite_ working yet).

## How to run

1. Construct the graph
    ```shell
    pip install -e .[dkg-construct]
    python -m mira.dkg.construct
    ```
    This puts all of the data into `$PYSTOW_HOME/mira/demo/import` (you only need to know this for debugging purposes) and pre-populates the `$NEO4J_HOME/data/` directory with a graph. See [here](https://neo4j.com/docs/operations-manual/current/configuration/file-locations/) for where `$NEO4J_HOME` is on each platform (e.g., it's at `/usr/local/var/neo4j` for me on mac).
2. Edit the `docker-compose.yml` to better reflect where your `$NEO4J_HOME` is in the volumes mapping for the neo4j service
2. Run the docker-compose configuration
    ```shell
    docker-compose up
    ```
    This runs the neo4j docker image (frontend at port 8769 and bolt port at 8770 so as to not conflict with any other neo4j instances running on the same machine), which should serve from the mapped volumes. It also runs the web frontend at port 8771, whose homepage gives a random term and `/ground/<text>` endpoint can be used to run grounding with Gilda. FYI, I already annotated these as the 3 ports I'll use on the CWC machine in the `/data/services/PORTS` file

## Todo

I'm having problems getting the actual database to load. It's strange since the dockerized neo4j is able to tell there are several databases, it's able to connect to the default one (note the community edition only allows you to connect to the one specified as "default", which is also inthe docker-compose.yml configuration), but it isn't recognizing data from inside it.

### CODE REVIEW

File: mira/dkg/construct.py [hunk 1]
```diff
@@ -183,8 +190,10 @@ def load():
     click.secho(command, fg="blue")
     os.system(command)  # noqa:S605
 
-    time.sleep(10)
-    os.system("brew services restart neo4j")
+    if restart:
+        time.sleep(10)
+        click.secho("restarting neo4j...")
+        os.system("brew services restart neo4j")
```
Comment (bgyori) [inline mira/dkg/construct.py:196]: Using `brew` will not work generally. Both on Mac and Linux, however, the `neo4j restart` command works as expected, at least for me so perhaps we should use that here.

**Comment by bgyori** [comment]
@cthoyt please remove all the Python code/comments related to loading the data into neo4j and managing a neo4j instance (configuration, restart etc.). These are handled as part of the Docker build and deployment now. The Python code for construction should end with a nodes.tsv.gz and edges.tsv.gz being produced.

File: mira/dkg/construct.py (hunk 1 above)
Comment (cthoyt) [reply mira/dkg/construct.py]: Subsumed by https://github.com/indralab/mira/pull/13#issuecomment-1218746374

**Comment by cthoyt** [comment]
> @cthoyt please remove all the Python code/comments related to loading the data into neo4j and managing a neo4j instance (configuration, restart etc.). These are handled as part of the Docker build and deployment now. The Python code for construction should end with a nodes.tsv.gz and edges.tsv.gz being produced.

Done in 1de09e8

**Comment by bgyori** [comment]
Thanks, should be good to go now!

**Pull request merged**